core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -588,32 +590,39 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     withAttributeNamespaceCheck(workspaceAttributeNames) {
       withLibraryAttributeNamespaceCheck(libraryAttributeNames) {
         getWorkspaceContextAndPermissions(sourceWorkspaceName, SamWorkspaceActions.read).flatMap { permCtx =>
-          dataSource.inTransaction({ dataAccess =>
-            // get the source workspace again, to avoid race conditions where the workspace was updated outside of this transaction
-            withWorkspaceContext(permCtx.toWorkspaceName, dataAccess) { sourceWorkspaceContext =>
-              DBIO.from(samDAO.getResourceAuthDomain(SamResourceTypeNames.workspace, sourceWorkspaceContext.workspaceId, userInfo)).flatMap { sourceAuthDomains =>
-                withClonedAuthDomain(sourceAuthDomains.map(n => ManagedGroupRef(RawlsGroupName(n))).toSet, destWorkspaceRequest.authorizationDomain.getOrElse(Set.empty)) { newAuthDomain =>
-
-                  // add to or replace current attributes, on an individual basis
-                  val newAttrs = sourceWorkspaceContext.attributes ++ destWorkspaceRequest.attributes
-
-                  withNewWorkspaceContext(destWorkspaceRequest.copy(authorizationDomain = Option(newAuthDomain), attributes = newAttrs), dataAccess) { destWorkspaceContext =>
-                    dataAccess.entityQuery.copyAllEntities(sourceWorkspaceContext, destWorkspaceContext) andThen
-                      dataAccess.methodConfigurationQuery.listActive(sourceWorkspaceContext).flatMap { methodConfigShorts =>
-                        val inserts = methodConfigShorts.map { methodConfigShort =>
-                          dataAccess.methodConfigurationQuery.get(sourceWorkspaceContext, methodConfigShort.namespace, methodConfigShort.name).flatMap { methodConfig =>
-                            dataAccess.methodConfigurationQuery.create(destWorkspaceContext, methodConfig.get)
+          // if the source bucket is a regional bucket, retrieve the region as the destination bucket also needs to be created in the same region
+          val bucketLocationFuture: Future[Option[String]] = for {
```

Comment:
The rule of thumb I **try** to follow is to explicitly show the data types when the type can't be inferred by simply reading the code or if it would it make reading and understanding code easier without having to ask IntelliJ about it